tests/statistic_reporter_test.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import time
+from unittest.mock import MagicMock
+
+from forwarder.statistics_reporter import StatisticsReporter
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+def test_that_warning_logged_on_send_exception(caplog):
+    statistics_reporter = StatisticsReporter("localhost", logger)
+    statistics_reporter._sender = MagicMock()
+    statistics_reporter._sender.send.side_effect = ValueError
+    with caplog.at_level(logging.WARNING):
+        statistics_reporter.send_pv_numbers("NOTINTORFLOAT", time.time())
+    assert "Could not send statistic: " in caplog.text
+
+
+def test_that_send_called_only_after_update_intervals():
+    statistics_reporter = StatisticsReporter("localhost", logger)
+    statistics_reporter._sender = MagicMock()
+
+    # test that first message is sent
+    t1 = int(time.time())
+    statistics_reporter.send_pv_numbers(2, t1)
+    assert t1 == statistics_reporter._last_update_s
+
+    # Wait for update_interval / 2 seconds
```

Comment:
Repeats the code, better to express the intent, e.g. try sending before interval time is complete